src/plugins/api_tests/src/core/external_script/base_scheme_interface/CInterfaceSasTests.cpp

```diff
@@ -31,12 +32,32 @@
 
 #include "SchemeSimilarityUtils.h"
 
-static const QString WORKING_DIR = U2::AppContext::getWorkingDirectoryPath();
-#ifndef Q_OS_MAC
-static const QString PROPER_WD_SCHEMES_PATH = WORKING_DIR + "/../../test/_common_data/cmdline/wd-sas-schemes/";
-#else
-static const QString PROPER_WD_SCHEMES_PATH = WORKING_DIR + "/../../../../../../test/_common_data/cmdline/wd-sas-schemes/";
-#endif
+#include <QFileInfo>
+
+static QString getTestDirImpl() {
+    QString testDir = qgetenv("UGENE_TESTS_PATH");
+    auto getDefaultTestDir = []() -> QString {
+        QString defaultTestDir;
+        #ifndef Q_OS_MAC
+        defaultTestDir = U2::AppContext::getWorkingDirectoryPath() + "/../../test/";
+        #else
+        defaultTestDir = U2::AppContext::getWorkingDirectoryPath() + "/../../../../../../test/";
+        #endif
+        return defaultTestDir;
+    };
+    if (testDir.isEmpty()) {
+        testDir = getDefaultTestDir();
+    } else if (!QFileInfo::exists(testDir)) {
+        QString defaultTestDir = getDefaultTestDir();
+        U2::coreLog.error(QString("UGENE_TESTS_PATH doesn't exist: '%1'. The default path is set: '%2'.").arg(testDir).arg(defaultTestDir));
+        testDir = getDefaultTestDir();
+    }
+
+    testDir = testDir + (testDir.endsWith("/") ? "" : "/");
+    return testDir;
```

Comment:
Do you need to check that the directory has the appropriate permissions to work with it (read/write)?

Reply:
This is for testing,so this directory is set up by us. I think, permission check is excess